### PR TITLE
EVG-5524: Override Config Default Alias with a CLI Variant/Task Pair

### PR DESCRIPTION
**Pull request by ybrill**

### CODE REVIEW

File: operations/patch_util_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	}
+	conf, err := NewClientSettings(s.tempDir + testConfigFile)
+
+	pp.loadAlias(conf)
+	pp.loadVariants(conf)
+	pp.loadTasks(conf)
+
+	s.Equal("", pp.Alias)
+	s.Equal([]string{"myvariant1", "myvariant2"}, pp.Variants)
+	s.Equal([]string{"mytask1", "mytask2"}, pp.Tasks)
```
Comment (tychoish) [inline operations/patch_util_test.go]: Maybe do Contains assertions here

File: operations/patch_util_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	err := ioutil.WriteFile(s.tempDir+testConfigFile, []byte(fileContents), 0644)
+	s.Require().NoError(err)
+
+	pp := patchParams{
+		Project:     "mci",
+		Variants:    []string{"myvariant1", "myvariant2"},
+		Tasks:       []string{"mytask1", "mytask2"},
+		SkipConfirm: true,
+	}
+	conf, err := NewClientSettings(s.tempDir + testConfigFile)
```
Comment (tychoish) [inline operations/patch_util_test.go]: no error this?

File: operations/patch_util_test.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+	pp.loadAlias(conf)
+	pp.loadVariants(conf)
+	pp.loadTasks(conf)
+
+	s.Equal("", pp.Alias)
```
Comment (tychoish) [inline operations/patch_util_test.go]: I'd use an s.Zero() assertion here.

File: operations/patch_util_test.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	s.Equal([]string{"myvariant1", "myvariant2"}, pp.Variants)
+	s.Equal([]string{"mytask1", "mytask2"}, pp.Tasks)
+}
+
+func (s *PatchUtilTestSuite) TearDownSuite() {
+	os.RemoveAll(s.tempDir)
```
Comment (tychoish) [inline operations/patch_util_test.go]: This returns an error, which we may want to make an assertion about.

File: operations/patch_util_test.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	}
+	conf, err := NewClientSettings(s.tempDir + testConfigFile)
+
+	pp.loadAlias(conf)
+	pp.loadVariants(conf)
+	pp.loadTasks(conf)
+
+	s.Equal("testing", pp.Alias)
+	s.Equal([]string(nil), pp.Variants)
+	s.Equal([]string(nil), pp.Tasks)
```
Comment (tychoish) [inline operations/patch_util_test.go]: s.Nil()

File: operations/patch_util_test.go [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+	err := ioutil.WriteFile(s.tempDir+testConfigFile, []byte(fileContents), 0644)
+	s.Require().NoError(err)
+
+	pp := patchParams{
+		Project:     "mci",
+		Alias:       "testing",
+		SkipConfirm: true,
+	}
+	conf, err := NewClientSettings(s.tempDir + testConfigFile)
```
Comment (tychoish) [inline operations/patch_util_test.go]: check this error.

File: operations/patch_util_test.go [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+const testConfigFile = ".evergreen.yml"
```
Comment (tychoish) [inline operations/patch_util_test.go]: I'd just put this as a parameter that's set in setup suite.

File: operations/patch_util.go [hunk 8]
```diff
@@ -212,44 +221,9 @@ func (p *patchParams) validatePatchCommand(ctx context.Context, conf *ClientSett
...
-		if len(defaultTasks) == 0 && !p.SkipConfirm &&
-			confirm(fmt.Sprintf("Set %v as the default tasks for project '%v'?",
-				p.Tasks, p.Project), false) {
-			conf.SetDefaultTasks(p.Project, p.Tasks...)
-			if err := conf.Write(""); err != nil {
-				grip.Warningf("warning - failed to set default tasks: %v\n", err)
-			}
-		}
+	if (len(p.Tasks) == 0 || len(p.Variants) == 0) && p.Alias == "" && p.Finalize {
+		err = errors.Errorf("Need to specify at least one task/variant or alias when finalizing.")
```
Comment (tychoish) [inline operations/patch_util.go]: I'd just return this on the next line.

File: operations/patch_util.go [hunk 9]
```diff
@@ -183,7 +183,15 @@ func (p *patchParams) validatePatchCommand(ctx context.Context, conf *ClientSett
 		grip.Warningf("warning - failed to set default alias: %v\n", err)
 	}
 
-	// Validate the alias if it exists
+	if err = p.loadVariants(conf); err != nil {
+		grip.Warningf("warning - failed to set default variants: %v\n", err)
+	}
+
+	if err = p.loadTasks(conf); err != nil {
+		grip.Warningf("warning - failed to set default tasks: %v\n", err)
```
Comment (tychoish) [inline operations/patch_util.go]: shouldn't we return here?

Comment (ybrill) [reply operations/patch_util.go]: I don't think this error should not stop us from going forward. 
The error is just in writing the default alias/variant/task to the user's config file.

File: operations/patch_util.go (hunk 8 above)
Comment (ybrill) [reply operations/patch_util.go]: Isn't that what it's doing?

File: operations/patch_util_test.go (hunk 1 above)
Comment (ybrill) [reply operations/patch_util_test.go]: Should be good now.

File: operations/patch_util_test.go (hunk 2 above)
Comment (ybrill) [reply operations/patch_util_test.go]: Should be good now.

File: operations/patch_util_test.go (hunk 3 above)
Comment (ybrill) [reply operations/patch_util_test.go]: Done.

File: operations/patch_util_test.go (hunk 4 above)
Comment (ybrill) [reply operations/patch_util_test.go]: Why is this important to check? (I did put in the check now)

File: operations/patch_util_test.go (hunk 5 above)
Comment (ybrill) [reply operations/patch_util_test.go]: Done.

File: operations/patch_util_test.go (hunk 6 above)
Comment (ybrill) [reply operations/patch_util_test.go]: Done.

File: operations/patch_util_test.go (hunk 7 above)
Comment (ybrill) [reply operations/patch_util_test.go]: Done.

File: operations/patch_util.go (hunk 8 above)
Comment (tychoish) [reply operations/patch_util.go]: I'd just combine these things on two lines

**Pull request merged**